src/main/java/com/dao/LocalJsonMailDAO.java

```diff
@@ -0,0 +1,97 @@
+package com.dao;
+
+import com.dto.MailDto;
+import com.model.Mail;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class LocalJsonMailDAO implements MailDao {
+
+    private List<MailDto> data = new ArrayList<>();
+
+    @PostConstruct
+    public void init() {
+        load();
+    }
+
+    @PreDestroy
+    public void destroy() {
+        save();
+    }
+
+    @Override
+    public void add(MailDto mailDto) {
+        data.add(mailDto);
+    }
+
+    @Override
+    public List<Mail> getAll() {
+        return data.stream()
```

Comment:
You should save your domain entity, ie `Mail` to some storage and use a `MailDTO` to transfer data to and from the client. DTO = data transfer object, so it should be used to transfer, not to persist your data